### PR TITLE
FIx more miscellaneous pytests failures

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This PR fixes multiple issues:

- [x] Enables `corr` and `cov` for `Datetime` and `Timedelta` types.
- [x] Properly disables `all` and `any` for `StringColumn`.
- [x] Preserves groupby categorical index ordering.
- [x] Catches `FutureWarnings` in pytests.

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_dataframe.py [hunk 1]
```diff
@@ -10807,6 +10794,7 @@ def test_series_data_with_name_with_columns_matching():
 def test_series_data_with_name_with_columns_not_matching():
     gdf = cudf.DataFrame(cudf.Series([1], name=2), columns=[1])
     pdf = pd.DataFrame(pd.Series([1], name=2), columns=[1])
+    # import pdb;pdb.set_trace()
```
Comment (mroeschke) [inline python/cudf/cudf/tests/test_dataframe.py:10797]:
```suggestion
```

File: python/cudf/cudf/core/groupby/groupby.py [hunk 2]
```diff
@@ -646,15 +646,15 @@ def agg(self, func):
...
-                if isinstance(result._index, cudf.CategoricalIndex):
-                    # Needs re-ordering the categories in the order
-                    # they are after grouping.
-                    result._index = cudf.Index(
-                        result._index._column.reorder_categories(
-                            result._index._column._get_decategorized_column()
-                        ),
-                        name=result._index.name,
-                    )
+                # if isinstance(result._index, cudf.CategoricalIndex):
```
Comment (mroeschke) [inline python/cudf/cudf/core/groupby/groupby.py:649]: Still needed?

File: python/cudf/cudf/core/dataframe.py [hunk 3]
```diff
@@ -1038,7 +1038,7 @@ def _init_from_dict_like(
                 empty_column = functools.partial(
                     cudf.core.column.column_empty,
                     row_count=(0 if index is None else len(index)),
-                    dtype=None,
+                    # dtype=None,
```
Comment (mroeschke) [inline python/cudf/cudf/core/dataframe.py:1041]: Can be removed?

Comment (galipremsagar) [inline python/cudf/cudf/core/dataframe.py:1041]:
```suggestion
```

**Review by mroeschke — APPROVED** [review]
Minor stuff but LGTM

File: python/cudf/cudf/core/groupby/groupby.py (hunk 2 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/groupby/groupby.py]: Nope, removed

File: python/cudf/cudf/core/dataframe.py (hunk 3 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py]: Yes, removed.

**Pull request merged**